scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -219,7 +224,10 @@ class UDPPeerGroup[M](val config: Config)(
       }
     }
 
-    override def in: ConnectableObservable[ChannelEvent[M]] = messageSubject
+    //TODO how to finish stream from queue?
```

Comment:
I think the solution for that would be to:
1. Add a new `case class ChannelClosed(reason: cats.effect.ExitCase) extends ChannelEvent[Nothing]`
2. Instead of `repeatEvalF` use [unfoldEval](https://monix.io/api/current/monix/reactive/Observable$.html#unfoldEval[S,A](seed:=%3ES)(f:S=%3Emonix.eval.Task[Option[(A,S)]]):monix.reactive.Observable[A]), seeded with the queue, which would return `None` after the exit. Or use a custom subscriber that polls the queue in a loop and raises `onError` and `onComplete` appropriately, recursing when the `F[Ack]` returned by `onNext` is completed.

The problem is that if we emit an exit event, there can be strictly one consumer of the queue, or else only one of them gets it. With multiple consumers it would be up to the user to make sure the queue is relayed into a `ConcurrentChannel` (with consumers already set up) or there's one multicasting `Observable`.

Reply:
https://github.com/input-output-hk/scalanet/pull/80#issuecomment-677552959 handles the case of accidental multiple consumers as well, forget the event.